goldrush_path/goldrush_path.cpp

```diff
@@ -34,6 +35,64 @@
 #include <tuple>
 #include <vector>
 
+uint32_t
+calc_median(std::vector<uint32_t>& vec, const size_t n)
+{
+  // sort vector in descending order
+  std::sort(vec.begin(), vec.end(), std::greater<uint32_t>());
+  return vec[n / 2];
+}
+
+void
+log_phred_calculations(size_t num_reads, std::vector<uint32_t>& phred_scores)
+{
+  if (opt::debug) {
+    std::cerr << "Number of reads used to calculate median: " << num_reads
+              << std::endl;
+    std::cerr << "Median array: ";
+    for (const auto& phred_score : phred_scores) {
+      std::cerr << phred_score << " ";
+    }
+    std::cerr << std::endl;
+  }
+  if (opt::verbose) {
+    std::cerr << "Minimum phred score calculated with median: "
+              << opt::phred_min << std::endl;
+  }
+}
+
+void
+calc_min_phred_threshold()
+{
+  constexpr size_t MEDIAN_SAMPLES_NEEDED = 50000;

```

Comment:
Would have expected constants like this and `MINIMUM_PHRED_THRESHOLD` to be at the top of the file, vs. within a function? Would be easier to organize like that I think in case we ever want to change constants?